component.go

```diff
@@ -44,3 +45,18 @@ func (c Component) Validate() error {
 
 	return nil
 }
+
+func CopyComponents(components []Component) []Component {
```

Comment:
Why did you make this function public? It's only used in the same package or am I missing something?

Reply:
Right now it is only used internally but providing such a primitive to lib users felt like a good thing.

Reply:
Note there is also `CopyBundles` and there will be `CopyChangelogs` coming with  the next PRs.